### ISSUE TITLE
uncrustify version check fails on minor releases

### ISSUE THREAD

**Issue by mrshmllow** [neovim/neovim#23565]
### Problem

When running `make formatc`, i have uncounted this annoying error.

```
> make formatc
...
FAILED: src/nvim/CMakeFiles/uncrustify-version /home/marsh/Projects/neovim/build/src/nvim/CMakeFiles/uncrustify-version 
CMake Error at /home/marsh/Projects/neovim/cmake/CheckUncrustifyVersion.cmake:11 (message):
/marsh/Projects/neovim/src/uncrustify.cfg -P /home/marsh/Projects/neovim/cmake/CheckUncrustifyVersion.cmake
CMake Error at /home/marsh/Projects/neovim/cmake/CheckUncrustifyVersion.cmake:11 (message):
  Wrong uncrustify version! Required version is 0.76.0 but found
  0.76.0144596232
  ```
  
  ```sh
> uncrustify -v
Uncrustify-0.76.0-144-59fb6232b
```

### Steps to reproduce

edit a c source file
`make formatc`

### Expected behavior

minor version releases should be allowed to run

### Neovim version (nvim -v)

master 84378c4dd56db9846b70a333530505a8048bd26e

### Vim (not Nvim) behaves the same?

no clue

### Operating system/version

arch linux rolling

### Terminal name/version

kitty 0.28.1

### $TERM environment variable

xterm-kitty

### Installation

occurs on source

**Comment by clason** [comment]
That's not a "minor release", though...

**Comment by clason** [comment]
also not a "patch release" -- this is a prerelease build version string

**Comment by mrshmllow** [comment]
I guess? Just extremely annoying since it requires patching my own PKGBUILD to even get 0.76.X since no repo has it yet and then manually editing `uncrustify.cfg`

I investigated further, and building on tag `uncrustify-0.75.1` and editing `uncrustify.cfg` to read 
```cfg
# Uncrustify-0.75.0_f
```
to fake it and...
```sh
> uncrustify -v
Uncrustify-0.75.1
```
```
CMake Error at /home/marsh/Projects/neovim/cmake/CheckUncrustifyVersion.cmake:11 (message):
  Wrong uncrustify version! Required version is 0.75.0 but found 0.75.1
  ```

**Comment by clason** [comment]
> As said before editing the version requirement and using head works just fine.

Yes, currently, but there's no guarantee.

**Comment by mrshmllow** [comment]
very strange they would do that

**Comment by dundargoc** [comment]
> As said before editing the version requirement and using head works just fine.

No. It happened to work fine in this instance. If we remove this check, and someone gets a different formatting than the one on CI they will get even more confused.
